docs/src/locators.md

```diff
@@ -68,189 +68,167 @@ await page.GetByRole("button", new() { Name = "Sign in" }).ClickAsync();
-
-
-## Locating elements
-
-Playwright comes with multiple built-in ways to create a locator. To make tests resilient, we recommend prioritizing user-facing attributes and explicit contracts, and provide dedicated methods for them, such as [`method: Page.getByText`]. It is often convenient to use the [code generator](./codegen.md) to generate a locator, and then edit it as you'd like.
-
-```js
-await page.getByText('Log in').click();
-```
-```java
-page.getByText("Log in").click();
-```
-```python async
-await page.get_by_text("Log in").click()
-```
-```python sync
-page.get_by_text("Log in").click()
-```
 ```csharp
-await page.GetByText("Log in").ClickAsync();
-```
+var locator = page.FrameLocator("#my-frame")
+    .GetByRole("button", new() { Name = "Sign in" });
 
-If you absolutely must use CSS or XPath locators, you can use [`method: Page.locator`] to create a locator that takes a [selector](./selectors.md) describing how to find an element in the page.
-
-Note that all methods that create a locator, such as [`method: Page.getByLabel`], are also available on the [Locator] and [FrameLocator] classes, so you can chain them and iteratively narrow down your locator.
-
-```js
-const locator = page.frameLocator('#my-frame').getByText('Submit');
-await locator.click();
+await locator.ClickAsync();
 ```
 
-```java
-Locator locator = page.frameLocator("#my-frame").getByText("Submit");
-locator.click();
-```
+### Locate by role
 
-```python async
-locator = page.frame_locator("#my-frame").get_by_text("Submit")
-await locator.click()
-```
+The [`method: Page.getByRole`] locator reflects how users and assistive technology perceive the page, for example whether some element is a button or a checkbox. When locating by role, you should usually pass the accessible name as well, so that the locator pinpoints the exact element.
 
-```python sync
-locator = page.frame_locator("my-frame").get_by_text("Submit")
-locator.click()
-```
+For example, consider the following DOM structure.
 
-```csharp
-var locator = page.FrameLocator("#my-frame").GetByText("Submit");
-await locator.ClickAsync();
+```html
+<button>Submit</button>
+<input type="checkbox" id="newsletter" name="newsletter" checked>
 ```
 
-### Locate based on accessible attributes
-
-The [`method: Page.getByRole`] locator reflects how users and assistive technology perceive the page, for example whether some element is a button or a checkbox. When locating by role, you should usually pass the accessible name as well, so that locator pinpoints the exact element.
+You can locate each element by it's implicit role:
 
 ```js
 await page.getByRole('button', { name: /submit/i }).click();
 
-await page.getByRole('checkbox', { checked: true, name: "Check me" }).check();
+await page.getByRole('checkbox', { checked: true, name: "newsletter" }).check();
```

Comment:
Since it is already checked, we should uncheck instead :)

```suggestion
await page.getByRole('checkbox', { checked: true, name: "newsletter" }).uncheck();
```